x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1849,14 +1849,18 @@ protected LogicalPlan rule(LogicalPlan plan) {
             if (plan instanceof Project) {
                 Project p = (Project) plan;
                 List<Object> values = extractConstants(p.projections());
-                if (values.size() == p.projections().size() && !(p.child() instanceof EsRelation)) {
+                if (values.size() == p.projections().size() && !(p.child() instanceof EsRelation) &&
+                    (!(p.child() instanceof LocalRelation) || (p.child() instanceof LocalRelation &&
```

Comment:
The checks you added here and below are the same. Is it worth extracting these checks into a meaningfully named method and use that in both places?